code/datums/weather/weather_types/ash_storm.dm

```diff
@@ -75,15 +75,18 @@
 	sound_wi.stop()
 
 /datum/weather/ash_storm/proc/is_ash_immune(atom/L)
-	while(L && !isturf(L))
-		if(ismecha(L)) //Mechs are immune
+	if(L)
+		if(ismecha(L.loc)) //Mechs are immune
+			return TRUE
+		if(isspacepod(L.loc)) //spacepods also protect you
+			return TRUE
+		if(is_ash_immune(L.loc))
```

Comment:
Best to put the recursion after the full check so after line 89. Might also benefit from a small comment about it